dff/utils/benchmark/context_storage.py

```diff
@@ -0,0 +1,585 @@
+        Return context with `from_dialog_len`, `message_dimensions`, `misc_dimensions`.
+
+        Wraps :py:func:`~.get_context`.
+        """
+        return get_context(self.from_dialog_len, self.message_dimensions, self.misc_dimensions)
+
+    def sizes(self):
+        """
+        Return sizes of objects defined by this config.
+
+        :return:
+            A dictionary with 4 elements:
+                - "starting_context_size" -- size of a context with `from_dialog_len`.
+                - "final_context_size" -- size of a context with `to_dialog_len`.
+                  A context of this size will never actually be benchmarked.
+                - "misc_size" -- size of a misc field of a context.
+                - "message_size" -- size of a misc field of a message.
+        """
+        return {
+            "starting_context_size": asizeof.asizeof(self.get_context()),
+            "final_context_size": asizeof.asizeof(
+                get_context(self.to_dialog_len, self.message_dimensions, self.misc_dimensions)
+            ),
+            "misc_size": asizeof.asizeof(get_dict(self.misc_dimensions)),
+            "message_size": asizeof.asizeof(get_message(self.message_dimensions)),
+        }
+
+    def get_context_updater(self):
+        """
+        Return context updater function based on configuration.
+
+        :return:
+            A function that accepts a context, modifies it and returns it.
+            The updated context has `step_dialog_len` more labels, requests and responses,
+            unless such dialog len would be equal to `to_dialog_len` or exceed than it,
+            in which case None is returned.
+        """
+
+        def _context_updater(context: Context):
+            start_len = len(context.labels)
+            if start_len + self.step_dialog_len < self.to_dialog_len:
+                for i in range(start_len, start_len + self.step_dialog_len):
+                    context.add_label((f"flow_{i}", f"node_{i}"))
+                    context.add_request(get_message(self.message_dimensions))
+                    context.add_response(get_message(self.message_dimensions))
+                return context
+            else:
+                return None
+
+        return _context_updater
+
+
+class BenchmarkCase(BaseModel):
+    """
+    This class represents a benchmark case and includes
+    information about it, its configuration and configuration of a context storage to benchmark.
+    """
+
+    name: str
+    """Name of a benchmark case."""
+    db_factory: DBFactory
+    """DBFactory that specifies context storage to benchmark."""
+    benchmark_config: BenchmarkConfig = BenchmarkConfig()
+    """Benchmark configuration."""
+    uuid: str = Field(default_factory=lambda: str(uuid4()))
+    """Unique id of the case. Defaults to a random uuid."""
+    description: str = ""
+    """Description of the case. Defaults to an empty string."""
+
+    @staticmethod
+    def set_average_results(benchmark):
```

Comment:
Why does this method belongs th this class exactly? Why don't we make it a function?